tests/race_conditions/utils_for_race_tests.py

```
@@ -0,0 +1,332 @@
+        assert promise.done(), f"cancelled() is True but done() is False: {promise!r}"
+
+    if promise.unpacked_once():
+        assert promise.unpacked_once_or_done(), (
+            f"unpacked_once() is True but unpacked_once_or_done() is False: {promise!r}"
+        )
+
+    if promise.done():
+        try:
+            promise.result()
+        except PromiseNotDoneError:
+            raise AssertionError(f"done() is True but result() raised PromiseNotDoneError: {promise!r}") from None
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)